UnityProject/Assets/Scripts/UI/Objects/Medical/GUI_DNAConsole.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using UI.Core.Net.Elements.Dynamic.Spawned;
+using UI.Core.NetUI;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace UI.Objects.Medical
+{
+	public class GUI_DNAConsole : NetTab
+	{
+		public NetTMPInputField target;
+
+		public NetTMPInputField find;
+		public NetTMPInputField replace;
+
+		public DNAStrandList DNAStrandList;
+
+		public Transform Viewport;
+
+		public List<PlayerHealthData> DEBUG_Species = new List<PlayerHealthData>();
+
+		public List<MutationSO> DEBUG_Mutations = new List<MutationSO>();
+
+
+		public DNASpeciesList DNASpeciesList;
+
+		public DNAConsole DNAConsole;
+
+		public NetPageSwitcher NetPageSwitcher;
+
+		public NetCountdownTimer netCountdownTimer;
+
+		public EmptyItemList MutationChoice;
+
+		public MutationUnlockMiniGame MutationUnlockMiniGame;
+
+		public Dictionary<MutationSO, MutationChooseElement> LoadedMutationSO =
+			new Dictionary<MutationSO, MutationChooseElement>();
+
+		public bool RemovingMutation = false;
+
+		public void SetRemovingMutation(bool newValue)
+		{
+			RemovingMutation = newValue;
+		}
+
+
+		public void Close()
+		{
+			ControlTabs.CloseTab(Type, Provider);
+		}
+
+		public void GenerateDNA()
+		{
+			var DNAMutationData = new DNAMutationData();
+
+			DNAMutationData.BodyPartSearchString = target.Value;
+
+
+			for (int i = 0; i < Viewport.childCount; i++)
```

Comment:
wouldn't it be better to do this in reverse just in-case any of it's elements changes while this is running?

Reply:
it's only generating a DNA payload it shouldn't change over the course, it's all it's doing is grabbing the data, and it's not doing any Await things